### ISSUE TITLE
sqlalchemy.exc.InvalidRequestError: Session is already flushing

### ISSUE THREAD

**Issue by abitrolly** [release-monitoring/anitya#689]
Trying to login to local development server with FAS gives traceback.
```
Jan 07 11:03:31 anitya-dev.example.com flask[4015]: [root INFO] No pre-discovered information supplied.
Jan 07 11:03:31 anitya-dev.example.com flask[4015]: [root INFO] Performing discovery on http://abitrolly.id.fedoraproject.org/
Jan 07 11:03:33 anitya-dev.example.com flask[4015]: [root INFO] Received id_res response from https://id.fedoraproject.org/openid/ using association {HMAC-SHA1}{5c3331e9>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]: 192.168.121.99 - - [07/Jan/2019 11:03:33] "GET /complete/fedora/?janrain_nonce=2019-01-07T11%3A03%3A08ZUjPTMM&openid.>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]: [werkzeug INFO] 192.168.121.99 - - [07/Jan/2019 11:03:33] "GET /complete/fedora/?janrain_nonce=2019-01-07T11%3A03%3A0>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]: Traceback (most recent call last):
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 2309, in __call__
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return self.wsgi_app(environ, start_response)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 2295, in wsgi_app
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     response = self.handle_exception(e)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask_restful/__init__.py", line 269, in error_>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return original_handler(e)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 1741, in handle_exception
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     reraise(exc_type, exc_value, tb)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     raise value
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     response = self.full_dispatch_request()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_requ>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     rv = self.handle_user_exception(e)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask_restful/__init__.py", line 269, in error_>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return original_handler(e)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_except>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     reraise(exc_type, exc_value, tb)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     raise value
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_requ>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     rv = self.dispatch_request()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return self.view_functions[rule.endpoint](**req.view_args)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_flask/utils.py", line 43, in wrapper
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return func(backend, *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_flask/routes.py", line 23, in complete
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/actions.py", line 43, in do_complete
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     user = backend.complete(user=user, *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/backends/base.py", line 40, in comp>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return self.auth_complete(*args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/backends/open_id.py", line 177, in >
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/strategy.py", line 160, in authenti>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return backend.authenticate(*args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/backends/base.py", line 80, in auth>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return self.pipeline(pipeline, *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/backends/base.py", line 83, in pipe>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     out = self.run_pipeline(pipeline, pipeline_index, *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/backends/base.py", line 113, in run>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     result = func(*args, **out) or {}
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/pipeline/user.py", line 75, in crea>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     'user': strategy.create_user(**fields)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_core/strategy.py", line 53, in create_us>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return self.storage.user.create_user(*args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_sqlalchemy/storage.py", line 144, in cre>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return cls._new_instance(cls.user_model(), *args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_sqlalchemy/storage.py", line 58, in _new>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return cls._save_instance(model(*args, **kwargs))
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/social_sqlalchemy/storage.py", line 64, in _sav>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     cls._session().commit()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/scoping.py", line 153, in do
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     return getattr(self.registry(), name)(*args, **kwargs)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 954, in commit
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     self.transaction.commit()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 467, in commit
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     self._prepare_impl()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 447, in _prepa>
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     self.session.flush()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2313, in flush
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     self._flush(objects)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2340, in _flush
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     self.dispatch.before_flush(self, flush_context, objects)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/event/attr.py", line 246, in __call__
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     fn(*args, **kw)
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/devel/anitya/db/events.py", line 78, in check_user
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     if new_obj.social_auth.count() == 0:
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/dynamic.py", line 269, in count
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     sess = self.session
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/dynamic.py", line 243, in session
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     sess.flush()
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:   File "/home/vagrant/.virtualenvs/anitya/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2307, in flush
Jan 07 11:03:33 anitya-dev.example.com flask[4015]:     raise sa_exc.InvalidRequestError("Session is already flushing")
Jan 07 11:03:33 anitya-dev.example.com flask[4015]: sqlalchemy.exc.InvalidRequestError: Session is already flushing
```

**Comment by abitrolly** [comment]
I can not login with FAS, GitHub, Google. Last two endpoints are 404 and 401.

**Comment by Zlopez** [comment]
Already fixed in PR #669 
Waiting for review
